### DISCUSSION TITLE
Does the generation use a random seed?

### DISCUSSION THREAD

**Discussion by pbosh** [oobabooga/text-generation-webui#236] [Q&A]
Sometimes I get the same answer from the exact same generation (system test conditions). I'm aware that even with a random seed this can happen but I just want to clarify if the generations currently use a random seed? If not how can make it use random seed Eg. -1

It would be a good addition in the next update to have a seed field.

**Comment by BadisG** [comment]
I agree, having the choice to get a random seed or a fixed seed would help us greatly on tests. Having a fixed seed gives us the exact same output and is a great way to see reproducibility (Comparaison Lora VS no Lora for example)

On Stable diffusion you can have this choice and on llama.cpp too. That's why I believe it would be great to also get the choice here.

**Reply by BarfingLemurs** [reply]
use the debug-deterministic preset for tests.
presets are adjusting parameters.
select your lora from that page.

**Reply by BadisG** [reply]
> use the debug-deterministic preset for tests. presets are adjusting parameters. select your lora from that page.

Having a fixed seed is better for tests. The debug-deterministic uses a do_sample = False and gives really repetitives sentences and poor results in general.

A fixed seed gives the same result no matter the preset, so you can be in the "normal conditions" with good outputs for tests.

**Reply by BarfingLemurs** [reply]
I guess so, but you how hardcore people will be about not breaking seeds

**Reply by BadisG** [reply]
What do you mean?

**Reply by BarfingLemurs** [reply]
well, like stable diffusion, people will be upset when sharing images with the seeds that no longer work :)

**Reply by BadisG** [reply]
Lmao!! True true 😄 
But seriously, having a seed would be a great addition to the repository
